server/index.js

```diff
@@ -1,104 +1,151 @@
-app.use('/assets', express.static(path.join(__dirname, '../public')))
+  app.use(userInfo)
 
-// strip trailing slashes from URLs
-app.get(/(.+)\/$/, (req, res, next) => {
-  res.redirect(req.params[0])
-})
+  // serve all files in the public folder
+  app.use('/assets', express.static(path.join(__dirname, '../public')))
 
-app.get('/view-on-site/:docId', (req, res, next) => {
-  const {docId} = req.params
-  const doc = getMeta(docId)
+  // strip trailing slashes from URLs
+  app.get(/(.+)\/$/, (req, res, next) => {
+    res.redirect(req.params[0])
+  })
 
-  if (!doc) return next(Error('Not found'))
+  app.get('/view-on-site/:docId', (req, res, next) => {
+    const { docId } = req.params
+    const doc = getMeta(docId)
 
-  res.redirect(doc.path)
-})
+    if (!doc) return next(Error('Not found'))
 
-// main pages
-app.use(readingHistory.middleware)
+    res.redirect(doc.path)
+  })
 
-// don't cache pages client-side to ensure browser always gets latest revision
-app.use((req, res, next) => {
-  res.set('Cache-Control', 'no-cache')
-  next()
-})
+  // main pages
+  app.use(readingHistory.middleware)
 
-// treat requests ending in .json as application/json
-app.use((req, res, next) => {
-  if (req.path.endsWith('.json')) {
-    req.headers.accept = 'application/json'
-    req.url = req.baseUrl + req.path.slice(0, -5)
-  }
-  next()
-})
+  // don't cache pages client-side to ensure browser always gets latest revision
+  app.use((req, res, next) => {
+    res.set('Cache-Control', 'no-cache')
+    next()
+  })
 
-app.use(pages)
-app.use(cache)
+  // treat requests ending in .json as application/json
+  app.use((req, res, next) => {
+    if (req.path.endsWith('.json')) {
+      req.headers.accept = 'application/json'
+      req.url = req.baseUrl + req.path.slice(0, -5)
+    }
+    next()
+  })
 
-// category pages will be cache busted when their last updated timestamp changes
-app.use(categories)
-app.use(playlists)
+  app.use(pages)
+  app.use(cache)
 
-postload.forEach((middleware) => app.use(middleware))
+  // category pages will be cache busted when their last updated timestamp changes
+  app.use(categories)
+  app.use(playlists)
 
-// if no page has been served, check for a redirect before erroring
-app.use(redirects)
+  postload.forEach((middleware) => app.use(middleware))
 
-// error handler for rendering the 404 and 500 pages, must go last
-app.use(errorPages)
+  // if no page has been served, check for a redirect before erroring
+  app.use(redirects)
 
-// If we are called directly, listen on port 3000, otherwise don't
+  // error handler for rendering the 404 and 500 pages, must go last
+  app.use(errorPages)
 
-if (require.main === module) {
-  app.listen(parseInt(process.env.PORT || '3000', 10))
+  // If we are called directly, listen on port 3000, otherwise don't
+
+  if (require.main === module) {
+    app.listen(parseInt(process.env.PORT || '3000', 10))
+  }
+
+  module.exports = app
 }
 
-module.exports = app
+(async () => {
```

Comment:
I'd suggest this file might be easiest to read top-down, in which case...

could we put this section at the top? and then have this be more or less:

getSecrets -> startServer

Reply:
Done